### PR TITLE
Switch watermill kafka to asynq redis

### PR DESCRIPTION
**Pull request by noboruma**

### CODE REVIEW

File: deepfence_worker/worker.go
Comment (gnmahanth) [inline deepfence_worker/worker.go:1]: remove these line, they are not required anymore
```
	// create if any topics is missing
	err := utils.CreateMissingTopics(
		cfg.KafkaBrokers, utils.Tasks,
		cfg.KafkaTopicPartitionsTasks, cfg.KafkaTopicReplicas, cfg.KafkaTopicRetentionMs,
	)
	if err != nil {
		log.Error().Msgf("%v", err)
	}
```

**Review by ramanan-ravi — APPROVED** [review]

File: deepfence_utils/directory/worker.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+func init() {
+	worker_clients_pool = sync.Map{}
+}
+
+func new_asynq_client(endpoints DBConfigs) (*asyncq_clients, error) {
+	if endpoints.Redis == nil {
+		return nil, errors.New("No defined Redis config")
+	}
+	redisCfg := asynq.RedisClientOpt{Addr: endpoints.Redis.Endpoint}
+	return &asyncq_clients{
```
Comment (ramanan-ravi) [inline deepfence_utils/directory/worker.go:37]: Can you add config?
```
asynq.Config{
            // Specify how many concurrent workers to use
            Concurrency: 50,
}
```

File: deepfence_utils/directory/worker.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+func init() {
+	worker_clients_pool = sync.Map{}
+}
+
+func new_asynq_client(endpoints DBConfigs) (*asyncq_clients, error) {
+	if endpoints.Redis == nil {
+		return nil, errors.New("No defined Redis config")
+	}
+	redisCfg := asynq.RedisClientOpt{Addr: endpoints.Redis.Endpoint}
```
Comment (ramanan-ravi) [inline deepfence_utils/directory/worker.go:36]: Can you add config?
```
asynq.Config{
            // Specify how many concurrent workers to use
            Concurrency: 50,
}
```

File: deepfence_worker/worker.go [hunk 3]
```diff
@@ -271,84 +112,69 @@ func startWorker(wml watermill.LoggerAdapter, cfg config) error {
...
 
-	log.Info().Msg("Starting the consumer")
-	if err = worker.Run(context.Background()); err != nil {
-		cancel()
-		return err
-	}
-	cancel()
-	return nil
+	log.Info().Msg("Starting the worker")
+	err = worker.Run(context.Background())
```
Comment (gnmahanth) [inline deepfence_worker/worker.go:178]: err is undefined var, it not defined anywhere in start_worker function

File: deepfence_worker/worker.go [hunk 4]
```diff
@@ -238,28 +88,19 @@ func startWorker(wml watermill.LoggerAdapter, cfg config) error {
...
-	publisher, err := kafka.NewPublisher(
-		kafka.PublisherConfig{
-			Brokers:   cfg.KafkaBrokers,
-			Marshaler: kafka.DefaultMarshaler{},
+	mux := asynq.NewServeMux()
+
+	srv := asynq.NewServer(
+		asynq.RedisClientOpt{Addr: cfg.RedisAddr},
+		asynq.Config{
+			Concurrency: 10,
```
Comment (ramanan-ravi) [inline deepfence_worker/worker.go:96]: Can you add ErrorHandler, Logger?
https://github.com/hibiken/asynq/blob/master/server.go#L155
```
	srv := asynq.NewServer(
		asynq.RedisClientOpt{Addr: cfg.RedisAddr},
		asynq.Config{
			Concurrency: 20,
			Queues: map[string]int{
				"critical": 6,
				"default":  3,
				"low":      1,
			},
			StrictPriority: false,
			ErrorHandler:   nil,
			Logger:         nil,
			LogLevel:       0,
		},
	)
```

File: deepfence_worker/main.go [hunk 5]
```diff
@@ -33,6 +33,7 @@ type config struct {
 	KafkaTopicReplicas        int16    `default:"1" split_words:"true"`
 	KafkaTopicRetentionMs     string   `default:"86400000" split_words:"true"`
 	KafkaTopicPartitionsTasks int32    `default:"3" split_words:"true"`
+	RedisAddr                 string   `default:"deepfence-redis:6379"`
```
Comment (gnmahanth) [inline deepfence_worker/main.go:36]: we use RedisHost, RedisPort, RedisDb and RedisPassword can we follow the same

**Review by gnmahanth — APPROVED** [review]

**Pull request merged**